### PR TITLE
Update entity alias creation to use entity lookup api

### PR DESCRIPTION
**Pull request by mbillow**
When creating an entity alias, the Vault Terraform Provider is currently listing all existing entities, making individual GET requests to those entities, and then inspecting the responses to make sure you aren’t making a duplicate alias.

Vault exposes a entity lookup API that we can leverage to accomplish this same duplication check in a single API call.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
- `resource/vault_identity_entity_alias`: Leverage entity lookup API to handle duplication pre-flight check instead of listing and retrieving all entities.
```

Output from acceptance testing:

```
terraform-provider-vault on  entity-alias-lookup [!] via 🐹 v1.17.6 
➜ TESTARGS="--run TestAccIdentityEntityAlias" make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v --run TestAccIdentityEntityAlias -timeout 30m ./...
?       github.com/hashicorp/terraform-provider-vault   [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/coverage      [no test files]
?       github.com/hashicorp/terraform-provider-vault/cmd/generate      [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/codegen   0.845s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/generated [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/datasources/transform/decode    1.665s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/datasources/transform/encode    0.585s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/alphabet    1.870s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/role        2.074s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/template    1.213s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/generated/resources/transform/transformation      1.452s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/helper    [no test files]
?       github.com/hashicorp/terraform-provider-vault/internal/consts   [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/internal/identity/entity  2.706s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/internal/identity/group   [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/internal/provider 0.866s [no tests to run]
?       github.com/hashicorp/terraform-provider-vault/schema    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/testutil  2.125s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/hashicorp/terraform-provider-vault/util      1.582s [no tests to run]
=== RUN   TestAccIdentityEntityAlias
--- PASS: TestAccIdentityEntityAlias (52.89s)
=== RUN   TestAccIdentityEntityAliasDuplicateFlow
--- PASS: TestAccIdentityEntityAliasDuplicateFlow (134.86s)
=== RUN   TestAccIdentityEntityAlias_Update
--- PASS: TestAccIdentityEntityAlias_Update (52.59s)
=== RUN   TestAccIdentityEntityAlias_Metadata
--- PASS: TestAccIdentityEntityAlias_Metadata (52.21s)
PASS
ok      github.com/hashicorp/terraform-provider-vault/vault     293.958s

```

### CODE REVIEW

File: vault/resource_identity_entity_alias.go [hunk 1]
```diff
@@ -75,32 +76,22 @@ func identityEntityAliasCreate(ctx context.Context, d *schema.ResourceData, meta
 
 	diags := diag.Diagnostics{}
 
-	var duplicates []string
-
 	mountAccessor := data["mount_accessor"].(string)
-	aliases, err := entity.FindAliases(client, &entity.FindAliasParams{
```
Comment (mbillow) [inline vault/resource_identity_entity_alias.go:81]: This is the only other use of this function is [vault/resource_identity_entity_alias_test.go#L191](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_identity_entity_alias_test.go#L191). I'm not sure if it is better to just replace that reference with a more specific helper function or just leave it as is.

The current implementation of `entity.FindAliases` is only really dangerous when running against a server with a large number of entities, which won't be the case when running the tests.

Comment (benashz) [reply vault/resource_identity_entity_alias.go:81]: Good point, we will take a closer look after your PR is merged!

File: vault/resource_identity_entity_alias.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package vault
 import (
 	"context"
 	"fmt"
+	"github.com/hashicorp/vault/api"
```
Comment (benashz) [inline vault/resource_identity_entity_alias.go:6]: nit: we prefer the that the built-in imports are always grouped together, we can update this after merge however.

**Review by benashz — APPROVED** [review]
LGTM!

Thank your contribution to HashiCorp!

**Pull request merged**